persist-cli/src/main/java/io/ballerina/persist/cmd/Generate.java

```diff
@@ -134,17 +144,74 @@ public void execute() {
+                        return;
+                    }
+                    generatedSourceDirPath = Paths.get(this.sourcePath, BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY,
+                            submodule);
+                }
+                if (!Files.exists(generatedSourceDirPath)) {
+                    try {
+                        if (!persistConfig.get("module").equals(packageName) && !Files.exists(
+                                Paths.get(this.sourcePath, BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY))) {
+                            Files.createDirectory(Paths.get(this.sourcePath,
+                                    BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY).toAbsolutePath());
+
+                        }
+                        Files.createDirectory(generatedSourceDirPath.toAbsolutePath());
+                    } catch (IOException e) {
+                        errStream.println("ERROR: failed to create the generated directory. " + e.getMessage());
+                        return;
+                    }
+                }
+
+                Path databaseConfigPath = generatedSourceDirPath.resolve(PATH_CONFIGURATION_BAL_FILE);
+                if (!Files.exists(databaseConfigPath)) {
+                    try {
+                        generateConfigurationBalFile(generatedSourceDirPath);
+                        errStream.printf(
+                                "Created database_configurations.bal file inside `%s` module in generated directory.%n",
+                                fileName.equals(packageName) ? "default" : fileName);
+                    } catch (BalException e) {
+                        errStream.println("ERROR: failed to generate the database_configurations.bal file. "
+                                + e.getMessage());
+                        return;
+                    }
+                }
+
+                if (!Files.exists(Paths.get(this.sourcePath, CONFIG_SCRIPT_FILE).toAbsolutePath())) {
+                    createConfigTomlFile(schemaFilePaths, packageName);
+                } else {
+                    updateConfigTomlFile(schemaFilePaths, packageName);
+                }
+
                 if (entityModule.getEntityMap().isEmpty()) {
                     errStream.printf("ERROR: the model definition file(%s) does not contain any entity definition.%n",
                             file.getFileName());
                     return;
                 }
-                if (entityModule.getModuleName().equals(packageName)) {
-                    generatedSourceDirPath = Paths.get(this.sourcePath, BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY);
-                } else {
-                    generatedSourceDirPath = Paths.get(this.sourcePath, BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY,
-                            entityModule.getModuleName());
-                }
                 if (!Files.exists(generatedSourceDirPath)) {
```

Comment:
We don't need to validate, as we are created generated source dir path here. passing error message is not correct

Reply:
But it's possible for the user to run the generate command twice right?

Reply:
Sorry, I didn't get your point here.

Reply:
this was a mistake and removed in [4365776](https://github.com/ballerina-platform/persist-tools/pull/148/commits/4365776aaf235f2a310b3750e72b217f87a574a6)